nomad/structs/structs.go

```diff
@@ -73,6 +73,9 @@ const (
 	ACLTokenUpsertRequestType
 	ACLTokenDeleteRequestType
 	ACLTokenBootstrapRequestType
+
+	// Constant for restart events that are within policy
+	ReasonWithinPolicy = "Restart within policy"
```

Comment:
I would move this maybe to be next to the RestartPolicy struct. We try to group vars/consts logically together. Also is there a need to duplicate this? Why not just use the client version.

Reply:
Got an import cycle when using client.ReasonWithinPolicy so I duplicated. Didn't seem worth it to try to break the import cycle since this is one constant.